scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SessionListener.java

```diff
@@ -62,4 +64,13 @@ void sessionEstablished(Handshaker handshaker, DTLSSession establishedSession)
 	 * @param peer the IP address and port of the peer the handshake has been completed with
 	 */
 	void handshakeCompleted(InetSocketAddress peer);
+
+	/**
+	 * Indicates that a handshake with a given peer has failed.
+	 * 
+	 * @param peer the IP address and port of the peer the handshake has failed
+	 * @param exception the occurred handshake exception, which caused the handshake to fail
+	 */
+	void handshakeFailed(InetSocketAddress peer, HandshakeException exception);
```

Comment:
The handshake may as well fail due to any other exception, don't you think? Is there anything specific we need to to with the `HandshakeException`, i.e. is it important that it is a `HandshakeException` and not a plain `Throwable`?